packages/cli/lib/supportHyperlinks.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+const supportsColor = require('supports-color');
+const hasFlag = require('has-flag');
+
+//See https://github.com/jamestalmage/supports-hyperlinks (License: https://github.com/jamestalmage/supports-hyperlinks/blob/master/license)
+
+function parseVersion(versionString) {
+  if (/^\d{3,4}$/.test(versionString)) {
+    // Env var doesn't always use dots. example: 4601 => 46.1.0
+    const m = /(\d{1,2})(\d{2})/.exec(versionString);
+    return {
+      major: 0,
+      minor: parseInt(m[1], 10),
+      patch: parseInt(m[2], 10),
+    };
+  }
+
+  const versions = (versionString || '').split('.').map(n => parseInt(n, 10));
+  return {
+    major: versions[0],
+    minor: versions[1],
+    patch: versions[2],
+  };
+}
+
+function supportsHyperlink(stream) {
+  const { env } = process;
+
+  if ('FORCE_HYPERLINK' in env) {
+    return !(
+      env.FORCE_HYPERLINK.length > 0 && parseInt(env.FORCE_HYPERLINK, 10) === 0
+    );
+  }
+
+  if (
+    hasFlag('no-hyperlink') ||
+    hasFlag('no-hyperlinks') ||
+    hasFlag('hyperlink=false') ||
+    hasFlag('hyperlink=never')
+  ) {
+    return false;
+  }
+
+  if (hasFlag('hyperlink=true') || hasFlag('hyperlink=always')) {
+    return true;
+  }
+
+  // If they specify no colors, they probably don't want hyperlinks.
+  if (!supportsColor.supportsColor(stream)) {
```

Comment:
IMO we should treat "supports color" and "supports hyperlinks" separately

Reply:
I'm going to disagree a bit here to say that I think this is a reasonable check. I have, however, added another separate check for colors in the new `getTerminalUISupport` util function, which I feel is a good compromise.